src/app/interfaces/strongRoute.ts

```diff
@@ -55,49 +72,47 @@ export class StrongRoute {
     const [full, parameters] = this.rootToHere();
     this.full = full;
     this.parameters = parameters;
-
-    if (full.length > 1) {
-      this.fullRoute = this.toRoute().join("/");
-    } else if (full.length === 1) {
-      this.fullRoute = this.name;
-    } else {
-      this.fullRoute = undefined;
-    }
-
-    this.config = { path: this.fullRoute, pathMatch: "full", ...config };
+    this.config = {
+      // Remove initial '/' from route path
+      path: this.toString().substr(1),
+      pathMatch: "full",
+      ...config,
+    };
   }
 
   /**
    * A base level (root) route component
    */
   public static get base() {
-    return new StrongRoute(null, null, {});
+    return new StrongRoute();
   }
 
   /**
    * Add a child route
    *
    * @param name Route name
+   * @param qsp Route query parameters
    * @param config Additional router configurations
    */
-  public add(name: string, config: Partial<Route> = {}) {
-    return new StrongRoute(this, name, config);
+  public add(name: string, qsp?: QSP, config?: Partial<Route>) {
+    return new StrongRoute(this, name, qsp, config);
   }
 
   /**
    * Add a new feature module route (inherit parent path without recalculating parent routes)
    *
    * @param name Route name
+   * @param qsp Route query parameters
    * @param config Additional router configurations
    */
-  public addFeatureModule(name: string, config: Partial<Route> = {}) {
-    return new StrongRoute(this, name, config, true);
+  public addFeatureModule(name: string, qsp?: QSP, config?: Partial<Route>) {
+    return new StrongRoute(this, name, qsp, config, true);
   }
 
   /**
    * Method used for templating a route with parameters.
    * Use this in a template like so:
-   * <a [routerlink]="route.Format({projectId: 1, siteId: 1})" />
+   * <a [routerlink]="route.format({projectId: 1, siteId: 1})" />
    */
   public format(args: { [key: string]: string | number }): string {
```

Comment:
I would like to see this function merged into `toRoute()` and `toString()` as both are effectively useless if the StrongRoute has parameters

Reply:
Yeah, no.

Format returns a whole, templated route (i.e. no missing parameters). It serves a distinct purpose, different from the other two methods

Reply:
Example output:

`/projects/123/sites/456?filter_name=abc`

Reply:
Add argument to accept QSPs?